addons/binding/org.openhab.binding.unifi/src/main/java/org/openhab/binding/unifi/internal/UniFiController.java

```diff
@@ -0,0 +1,403 @@
+import static java.net.HttpURLConnection.*;
+
+import java.io.IOException;
+import java.net.CookieManager;
+import java.net.SocketException;
+import java.net.SocketTimeoutException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.security.SecureRandom;
+import java.util.Calendar;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.BooleanUtils;
+import org.apache.commons.lang.StringUtils;
+import org.openhab.binding.unifi.internal.api.UniFiClient;
+import org.openhab.binding.unifi.internal.api.UniFiDevice;
+import org.openhab.binding.unifi.internal.api.UniFiSite;
+import org.openhab.binding.unifi.internal.api.UniFiWiredClient;
+import org.openhab.binding.unifi.internal.api.UniFiWirelessClient;
+import org.openhab.binding.unifi.internal.api.json.UniFiClientResponse;
+import org.openhab.binding.unifi.internal.api.json.UniFiDeviceResponse;
+import org.openhab.binding.unifi.internal.api.json.UniFiSiteResponse;
+import org.openhab.binding.unifi.internal.ssl.UniFiHostnameVerifier;
+import org.openhab.binding.unifi.internal.ssl.UniFiTrustManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ *
+ * The {@link UniFiController} is the main communication point with an external instance of the Ubiquiti Networks
+ * Controller Software.
+ *
+ * @author Matthew Bowman - Initial contribution
+ */
+public class UniFiController {
+
+    private static final Map<String, String> INSIGHTS_WITHIN_24H = Collections.singletonMap("within", "24");
+
+    private Logger logger = LoggerFactory.getLogger(UniFiController.class);
+
+    private UniFiControllerConfig config;
+
+    private HostnameVerifier hostnameVerifier = new UniFiHostnameVerifier();
+
+    private SSLSocketFactory sslSocketFactory;
+
+    private Map<String, UniFiSite> sitesCache = Collections.emptyMap();
+
+    private Map<String, UniFiDevice> devicesCache = Collections.emptyMap();
+
+    private Map<String, UniFiClient> clientsCache = Collections.emptyMap();
+
+    private Map<String, UniFiClient> insightsCache = Collections.emptyMap();
+
+    private ObjectMapper objectMapper = new ObjectMapper();
+
+    private CookieManager cookieManager = new CookieManager();
+
+    public UniFiController(UniFiControllerConfig config) throws UniFiException {
+        try {
+            SSLContext sc = SSLContext.getInstance("SSL");
+            sc.init(null, new TrustManager[] { new UniFiTrustManager() }, new SecureRandom());
+            sslSocketFactory = sc.getSocketFactory();
+        } catch (Exception e) {
```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception

Reply:
Honestly, I couldn't think of any "good" way to handle this as the UniFi controller uses a self-signed SSL certificate and if any of this logic fails (installing the blind trust manager), the binding won't function at all. 

I decided the best course of action was to catch Exception and rethrow a UniFiException to signal the issue in the PaperUI as a `ThingStatusDetail.CONFIGURATION_ERROR`.

Reply:
But what exceptions can come from the block given you implemented some accept all logic for the certificates